app/models/profile_field.rb

```diff
@@ -35,4 +38,10 @@ def type
   def generate_attribute_name
     self.attribute_name = label.titleize.scan(WORD_REGEX).join.underscore
   end
+
+  def maximum_header_field_count
+    return if self.class.header.count < HEADER_FIELD_LIMIT
```

Comment:
We want a maximum of 3 profile fields. So we return here as long as well still have less than that. If we changed the `<` to `<=` then we'd return without adding an error when 3 fields already exist and add a 4th one.

Reply:
Ahh, okay. No, you understood correctly, but I feel like that order of operations isn't super clear.

Also, is there ever a reason to update them in place? Because this protects against creation after 2, which it seems is where you were coming from, but we can't update any of them when we're currently at the max:

```ruby
[13] pry(main)> ProfileField.header.count
=> 3
[14] pry(main)> ProfileField.last.validate!
ActiveRecord::RecordInvalid: Validation failed: Display area maximum number of header fields (3) exceeded
from /Users/jamie/.rvm/gems/ruby-2.7.2/gems/activerecord-6.1.4/lib/active_record/validations.rb:80:in `raise_validation_error'
```

Reply:
> Ahh, okay. No, you understood correctly, but I feel like that order of operations isn't super clear.

Agreed, but you hit on that in your other comment, Rubocop kinda forced me into a guard clause which really doesn't work too well here. 

Good point regarding the validation, while we don't generally mutate them in place it is possible via the admin UI, so there's a subtle bug lurking here. I'll set this PR back to draft and update the validation tomorrow, including a test based on the example above. Thanks!

Reply:
True, this was definitely part of that same confusion.